community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/LuceneDocumentStructure.java

```diff
@@ -258,7 +260,7 @@ protected AcceptStatus accept( BytesRef term ) throws IOException
         private final Field idField;
         private final Field idValueField;
 
-        private final Map<String,Field> valueFields = new HashMap<>();
+        private Field[] reusableValueFields = new Field[0];
```

Comment:
Objection your honor! Premature optimization.

(I at least had no idea of how hot that spot was after just having dipped into Lucene for the first time. Now we know)